### ISSUE TITLE
Screen Reader: Profile Subsections: Incorrect focus movement after requesting new OTP code

### ISSUE THREAD

**Issue by elisenwazuoke** [Expensify/App#77494]
## Action Performed:
Precondition: User is logged in

1. Using TalkBack on Android, open the app
2. Swipe right to focus the Account tab and double-tap to activate it
3. Swipe right to focus the Profile button and double-tap to activate it
4. Swipe right to focus the Contact methods menu item and double-tap to activate it
5. Swipe right to focus an email address that requires verification and double-tap to activate it
6. Swipe right to focus the "Didn't receive a code?" button and double-tap to activate it
7. Note where the focus lands

This is a global issue occurring in all OTP screens.

Other Occurrences:
1. Profile > Contact methods > Unverified email: focus resets to the Back button when "Request code in 00:xx" changes to "Didn't receive a magic code?"

## Expected Result:
After requesting a new OTP code, focus should move to the first OTP cell so users can enter the new code.

## Actual Result:
Focus resets to the Back button instead.
Users lose context and must manually navigate to the OTP field again.

## Area issue was found in:
Profile Subsections

## Failed WCAG checkpoints
WCAG 2.4.3 Focus Order
https://www.w3.org/WAI/WCAG22/Understanding/focus-order

## User impact:
People with mobility impairments who must rely on keyboard access for operating a page benefit from a logical, usable focus order. People with vision impairments or disabilities that make reading difficult can become disoriented when tabbing takes focus someplace unexpected or when they cannot easily find the content surrounding an interactive element. Only a small portion of the page may be visible to an individual using a screen magnifier at a high level of magnification. Such a user may interpret a field in the wrong context if the focus order is not logical.

## Suggested resolution:
Set focus to the first OTP field programmatically after requesting a new code.
https://reactnative.dev/docs/accessibilityinfo#setaccessibilityfocus
https://appt.org/en/docs/react-native/samples/accessibility-focus

## Environment(s)
Samsung Galaxy Note10 Plus - Android 12, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos
<img width="1179" height="2438" alt="Image" src="https://github.com/user-attachments/assets/c1ef90bf-86b7-491a-a290-59aed7bb13c0" />

https://github.com/user-attachments/assets/15821f6d-396e-45c2-b674-0070bb3b4155

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** NA
**Logs:** https://stackoverflow.com/c/expensify/questions/4856 NA
**Notes/Photos/Videos:** Any additional supporting documentation Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7008083
**Issue reported by:** NA
**Slack conversation:** NA
